### ISSUE TITLE
Start over onDone

### ISSUE THREAD

**Issue by runedrune** [phucgaoxam/slide_countdown_clock#1]
Hey, is there any way to get the countdown clock to start again when it hits 0? I've tried many different ways but it just seems to get stuck at 0 no matter what I do.

**Comment by phucgaoxam** [comment]
@runedrune can you please wait for a few days for the next update.
I will mention you after I'm done.

**Comment by runedrune** [comment]
Yeah sure man. No problem.

**Comment by phucgaoxam** [comment]
@runedrune or you can try setState (reset the widget itself)

**Comment by runedrune** [comment]
I tried that. Doesn't seem to work. In onDone I tried setstate and passed a new value to duration but it didn't work. Just stayed at 0. Can't figure out why.

**Comment by bawantha** [comment]
yes setState or any other state management method won't work timer build only once.

**Comment by tylerjin** [comment]
modify  SlideCountdownClockState 
`void _init() {
    timeLeft = widget.duration;
    var time = DateTime.now();
    var initStream =
        Stream<DateTime>.periodic(Duration(milliseconds: 1000), (_) {
      timeLeft -= Duration(seconds: 1);
      if (timeLeft.inSeconds == 0) {
        Future.delayed(Duration(milliseconds: 1000), () {
          if (widget.onDone != null) widget.onDone();
        });
      }
      return time;
    });

    print(timeLeft.inSeconds);
    timeStream = initStream
        .where((event) =>
            event.isBefore(time.add(timeLeft + Duration(seconds: 1))))
        .asBroadcastStream();
  }

  @override
  Widget build(BuildContext context) {
    _init();`
